### PR TITLE
Refactor Traits Collection

### PR DESCRIPTION
**Pull request by xQwexx**
I want to work on making the traits better, but first I wanted to tidy the current code.

### CODE REVIEW

File: src/trait_manager/index.ts [hunk 1]
```diff
@@ -62,12 +62,9 @@ export default class TraitManager extends Module<TraitManagerConfig & { pStylePr
    */
   constructor(em: EditorModel) {
     super(em, 'TraitManager', defaults);
-    const c = this.config;
     const model = new Model();
     this.model = model;
-    const ppfx = c.pStylePrefix;
-    this.types = { ...typesDef };
-    ppfx && (c.stylePrefix = `${ppfx}${c.stylePrefix}`);
+    this.types = typesDef;
```
Comment (artf) [inline src/trait_manager/index.ts:67]: Looks all legit, the only issue I see is with the removal of the that thing for `stylePrefix` which makes them render incorrectly. 
traits

Don't you see them in the same way?

Comment (xQwexx) [reply src/trait_manager/index.ts:67]: Yeah thanks for catching it I didn't realised that it changed, I updated the fix for it.

File: src/trait_manager/model/Traits.ts [hunk 2]
```diff
@@ -33,30 +37,25 @@ export default class Traits extends Collection<Trait> {
...
+        traits[i] = trait instanceof Trait ? trait : this.tf.build(trait);
+        traits[i].target = this.target;
       }
+      return super.add(traits, opt);
     }
+    const trait = models instanceof Trait ? models : this.tf.build(models as any);
+    trait.target = this.target;
 
-    return Collection.prototype.add.apply(this, [models as Trait[], opt]);
+    return super.add(trait, opt);
```
Comment (artf) [inline src/trait_manager/model/Traits.ts:59]: I'm not sure exactly what is happening here but with these changes to Traits, undo/redo on traits doesn't work properly (eg. I tried my latest fix [here](https://github.com/GrapesJS/grapesjs/issues/4972) and it works only if I rollback these changes)

**Review by artf — CHANGES_REQUESTED** [review]

File: src/trait_manager/model/Traits.ts (hunk 2 above)
Comment (xQwexx) [reply src/trait_manager/model/Traits.ts:61]: I added a test to cover the undo scenario and fixed it :) Sorry I couldn't have time earlier  :(

**Review by artf — APPROVED** [review]

**Comment by artf** [comment]
Awesome, thanks @xQwexx ❤️

**Pull request merged**